### PR TITLE
Blogpost: `The Easier Way to Create Pulumi Providers in Go`

### PR DESCRIPTION
**Pull request by dixler**
Blogpost on the Pulumi Go Provider SDK.

Fixes https://github.com/pulumi/home/issues/2465

TODO:
- [x] Content Review
- [x] meta_desc
- [x] Thumbnail

**Note to reviewers:** I'm looking to get a review of the first and then asking for a thumbnail before merging it. I notice that the `meta_desc` isn't showing up and I'll get that working first.

### CODE REVIEW

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+We are excited to announce that we've updated the Pulumi Provider Boilerplate to make this easier!
+
+**Note:** This update will not have any impact on existing providers that are utilizing an older version of the repository.
+
+This major update brings a wealth of usability improvements to the Pulumi Provider Boilerplate by incorporating our brand
+new Pulumi Go Provider SDK [(view on GitHub)](https://github.com/pulumi/pulumi-go-provider).
+Now, you can implement your provider in Go by defining the provider's metadata, resources, and functions, and then building the provider binary.
+The library takes care of the gRPC server for you and the Pulumi CLI takes care of the SDK code generation with the following command
+`pulumi package gen-sdk <path/to/provider-binary>`.
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:38]: GitHub won't let me make the suggestion, but I'd suggest ending `the following command` with a `:`, adding an empty line, and putting `pulumi package gen-sdk` on its own line (in a code fence) to call it out more visually.

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
+title: "The Easier Way to Create Pulumi Providers in Go"
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:2]: What do you think about "An Easier Way" instead? 

```suggestion
title: "An Easier Way to Create Pulumi Providers in Go"
```

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md (hunk 1 above)
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:38]: GitHub won't let me make the suggestion, but I'd suggest ending `the following command` with a `:`, adding an empty line, and putting `pulumi package gen-sdk` on its own line (in a code fence) to call it out more visually.

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+authors: ["kyle-dixler"]
+tags: ["go", "packages"]
+meta_desc: "We are thrilled to announce a major update to the Pulumi Provider Boilerplate that simplifies the provider development process."
+date: "2023-01-03"
+---
+
+## Background
+
+The Pulumi Provider Boilerplate [(view on GitHub)](https://github.com/pulumi/pulumi-provider-boilerplate) is a template repository
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:11]: Does the README need to be updated on this repo to align with the changes announced here?

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+new Pulumi Go Provider SDK [(view on GitHub)](https://github.com/pulumi/pulumi-go-provider).
+Now, you can implement your provider in Go by defining the provider's metadata, resources, and functions, and then building the provider binary.
+The library takes care of the gRPC server for you and the Pulumi CLI takes care of the SDK code generation with the following command
+`pulumi package gen-sdk <path/to/provider-binary>`.
+
+You can leave the heavy lifting to us and focus on the implementation details that matter to your organization.
+
+We encourage everyone to try the new authoring experience, including devs who may have previously found it challenging in comparison to the ease of program authorship.
+
+## Additional Example
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:44]: It'd be super cool if this were accompanied by a short video showing a minimal example of how this actually comes together.

**Review by cnunciato — COMMENTED** [review]
A few initial comments for ya.

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md (hunk 2 above)
Comment (dixler) [reply themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:2]: I think `The Easier Way` is a little more :hot_pepper:

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md (hunk 4 above)
Comment (dixler) [reply themes/default/content/blog/pulumi-go-boilerplate-v2/index.md]: I've added a code example copied from the boilerplate.

**Comment by scottslowe** [comment]
@aaronkao @cnunciato We should slot this into our content agenda, rather than merging it out of sequence. This will help with the planned blog content cadence.

**Comment by dixler** [comment]
@cnunciato you may need to update the publish date to get it where you want it

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+## New Provider Boilerplate
+
+We are excited to announce that we've updated the Pulumi Provider Boilerplate to make this easier!
+
+**Note:** This update will not have any impact on existing providers that are utilizing an older version of the repository.
+
+This major update brings a wealth of usability improvements to the Pulumi Provider Boilerplate by incorporating our brand
+new Pulumi Go Provider SDK [(view on GitHub)](https://github.com/pulumi/pulumi-go-provider).
+Now, you can implement your provider in Go by defining the provider's metadata, resources, and functions, and then building the provider binary.
```
Comment (AaronFriel) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:45]: Would like to work in the term "code-first" here, elaborating that we've replaced the need for providers to declare a separate `schema.json` for SDK codegen.

**Review by AaronFriel — COMMENTED** [review]
Not needing to hand-write the schema is _the_ feature we're delivering, would be good to elaborate on that. 

Really happy with the code samples chosen and the new boilerplate.

**Comment by dixler** [comment]
@AaronFriel made some changes. lmk if it looks ok.

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
+---
+title: The Easier Way to Create Pulumi Providers in Go
+allow_long_title: true
+date: 2023-01-03
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:4]:
```suggestion
date: 2023-01-19
```

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+meta_desc: A major update to the Pulumi Provider Boilerplate simplifies the provider development process.
+meta_image: meta.png
+---
+
+We are excited to announce that we've updated the Pulumi Provider Boilerplate to make custom provider implementation easier!
+
+**Note:** This update will not have any impact on existing providers that are utilizing an older version of the repository.
+
+This major update brings a wealth of usability improvements to the Pulumi Provider Boilerplate by incorporating our brand
+new Pulumi Go Provider SDK.
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:19]: I'd swap these:

```suggestion
This major update brings a wealth of usability improvements to the Pulumi Provider Boilerplate by incorporating our brand new Pulumi Go Provider SDK.

**Note:** This update will not have any impact on existing providers that are utilizing an older version of the repository.
```

**Review by cnunciato — APPROVED** [review]
Looks great from my perspective. :shipit:

File: themes/default/content/blog/pulumi-go-boilerplate-v2/index.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+<!--more-->
```
Comment (dixler) [inline themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:21]: @cnunciato how about

```suggestion
We are excited to announce that we've updated the Pulumi Provider Boilerplate to make custom provider implementation easier!

This major update brings a wealth of usability improvements to the Pulumi Provider Boilerplate by incorporating our brand
new Pulumi Go Provider SDK.

<!--more-->

**Note:** This update will not have any impact on existing providers that are utilizing an older version of the repository.
```

Comment (cnunciato) [reply themes/default/content/blog/pulumi-go-boilerplate-v2/index.md:21]: Perfect! (Although I'd love it even more without those hard line breaks, but I can live with it. 😄)

**Comment by cnunciato** [comment]
Merging on Kyle’s 👍.

**Pull request merged**